File: 7-reverse-integer/reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        if not x:
            return 0

        digit = []
        y = abs(x)

        while (y != 0):
            digit.append(str(y % 10))
            y = y // 10

        res = int("".join(digit)) if x >= 0 else -int("".join(digit))

        if res <= pow(2, 31) -1 and res >= -pow(2, 31):
            return res
        else:
            return 0


        

File: 7-reverse-integer/reverse-integer.cpp
using namespace std;

class Solution {
public:
    int reverse(int x) {
        double res = 0;
        while (x != 0) {
            res += x % 10;
            x /= 10;
            res *= 10;
        }
        res /= 10;
        return res <= pow(2, 31) - 1 && res >= -pow(2, 31)   ? (int) res : 0;
    }
};